webpack/assets/javascripts/react_app/components/ForemanModal/subcomponents/ForemanModalFooter.js

```diff
@@ -3,18 +3,32 @@ import PropTypes from 'prop-types';
 import { Modal, Button } from 'patternfly-react';
 import { useModalContext } from '../ForemanModalHooks';
 
+import SubmitOrCancel from './SubmitOrCancel';
+
 const ForemanModalFooter = props => {
   const childCount = React.Children.count(props.children);
-  const { onClose } = useModalContext();
+  const { onClose, isSubmitting, id, submitProps } = useModalContext();
+
   // Render the provided children, or default markup if none given
+  const closeButton = childCount === 0 && (
+    <Button bsStyle="default" onClick={onClose}>
+      Close
+    </Button>
+  );
+
+  const submitOrCancel = childCount === 0 && submitProps && (
+    <SubmitOrCancel
+      isSubmitting={isSubmitting}
+      onCancel={onClose}
+      submitProps={submitProps}
+      id={id}
```

Comment:
I'm thinking maybe `id` should be a required prop, to avoid undefined values being passed on.

Reply:
It gets pulled from modal context in `ForemanModalFooter` and `SubmitOrCancel` has it as a required prop.

Reply:
@xprazak2 sorry, I think I commented in the wrong place.  I meant the ID passed to the delete modal to tell it which model to delete.

Reply:
I see. The problem is that until the delete button is clicked, we do not know id of a model we want to delete, so we would get a failed prop type...